packages/@aws-cdk/aws-eks/README.md

```diff
@@ -397,6 +397,46 @@ new KubernetesPatch(this, 'hello-kub-deployment-label', {
 })
 ```
 
+### Querying Kubernetes Resources
+
+The `KubernetesGet` construct can be used to query for information about kubernetes resources,
+and use that as part of your CDK application.
+
+For example, you can fetch the address of a [`LoadBalancer`](https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer) type service:
+
+```typescript
+// query the load balancer address
+const myServiceAddress = new KubernetesGet(this, 'LoadBalancerAttribute', {
+  cluster: cluster,
+  resourceType: 'service',
+  resourceName: 'my-service',
+  jsonPath: '.status.loadBalancer.ingress[0].hostname', // https://kubernetes.io/docs/reference/kubectl/jsonpath/
+});
+
+// pass the address to a lambda function
+const proxyFunction = new lambda.Function(this, 'ProxyFunction', {
+  ...
+  environment: {
+    myServiceAddress: myServiceAddress.value
+  },
+})
+```
+
+#### Service Description
+
+The `ServiceDescription` construct can be used to quickly access information about services. Internally, it uses the `KubernetesGet` resource for all its attributes.
```

Comment:
Do we really need `ServiceDescription`?? I have to say that I am not super comfortable with this API for some reason. Maybe:

1. Prefix with `Kubernetes` to indicate it's a k8s service.
2. I don't like the `Description` postfix. I rather this will represent a _service_ and not the _service description_.
3. I feel this is a bit "over designed" and future proofs too much. The use case is "give me the LB address for a service inside my cluster", maybe we should just address this use case directly?


```ts
cluster.getLoadBalancerAddress('my-service', { namespace: 'default' });

// or query
cluster.queryLoadBalancerAddress('my-service', { namespace: 'default' });
```

Reply:
I have to say that as far as the API goes, I really like it, kind of rolls of your keyboard:

```typescript
const loadBalancerAddress = cluster.describeService('my-service').loadBalancerAddress
```

Adding a `Kubernetes` prefix sounds good, but i'm ok with the `Description` suffix since just using `Service` feels off because i think `Service` is mainly associated in people's head with the actual resource definition. Also, when people start using cdk8s here, it will *"conflict"* with the `kplus.Service` construct. 

`Description` also aligns with `kubectl describe` (which is what users use to query service information), and coincidentally also aligns with the `describeXXX` commands of the aws cli, so it feels familiar in the aws world.

Having said that, I agree it might be a bit overly designed, especially since i can't seem to think of any other attribute this `Description` might include. So i'm good with being more specific for now 👍 

i'll go with `getServiceLoadBalancerAddress`